src/hlstrafe.cpp

```diff
@@ -1351,9 +1351,13 @@ namespace HLStrafe
 				player.Velocity[1] = static_cast<float>(std::sin(player.Viewangles[1] * M_DEG2RAD) * std::cos(player.Viewangles[0] * M_DEG2RAD) * 350 * 1.6);
 			}
 		}
-		player.Velocity[2] = static_cast<float>(std::sqrt(2 * 800 * 45.0));
+		player.Velocity[2] = static_cast<float>(std::sqrt(2 * 800 * 45.0) * ((100.0 - (player.StaminaTime / 1000.0) * 19.0) / 100.0));
```

Comment:
This applies StaminaTime (which is the value of fuser2) unconditionally. However, other mods are free to put whatever unrelated value into fuser2, so there should be some sort of check to ignore it. Also below you set StaminaTime, also unconditionally, when it only applies to CS 1.6.

I'm not sure what the best way to do this, maybe a flag like HasStamina? Similar to the Bhopcap flag.

Reply:
Yeah, I can have the check similarly done in HwDLL and pass it into here through the MovementVars struct. I was really concerned with clogging the whole thing with conditional statements for just a single game but it seems like a okay.